jwql/website/apps/jwql/forms.py

```diff
@@ -51,22 +51,177 @@ def view_function(request):
 
-# from data_containers import get_thumbnails_all_instruments
 from jwql.database import database_interface as di
 from jwql.utils.constants import ANOMALY_CHOICES
+from jwql.utils.constants import ANOMALIES_PER_INSTRUMENT
 from jwql.utils.constants import FILTERS_PER_INSTRUMENT
 from jwql.utils.constants import FULL_FRAME_APERTURES
 from jwql.utils.constants import GENERIC_SUFFIX_TYPES
 from jwql.utils.constants import JWST_INSTRUMENT_NAMES_MIXEDCASE
 from jwql.utils.constants import JWST_INSTRUMENT_NAMES_SHORTHAND
 from jwql.utils.constants import OBSERVING_MODE_PER_INSTRUMENT
+from jwql.utils.constants import ANOMALY_CHOICES_PER_INSTRUMENT
 from jwql.utils.utils import get_config, filename_parser
-# from jwql.website.apps.jwql.views import current_anomalies  ### global variable defined once query_anomaly page has forms filled
+from jwql.utils.utils import query_format
+
+from wtforms import SubmitField, StringField
 
 FILESYSTEM_DIR = os.path.join(get_config()['jwql_dir'], 'filesystem')
 
-# from jwql.utils import anomaly_query_config
-# from jwql.website.apps.jwql import views # update anomaly_query_config
+
+class BaseForm(forms.Form):
+    """A generic form with target resolve built in"""
+    # Target Resolve
+    targname = StringField('targname', default='')
+    target_url = StringField('target_url', default='')
+
+    # Submit button
+    resolve_submit = SubmitField('Resolve Target')
+
+
+class DynamicAnomalyForm(BaseForm):
+    """Form validation for the anomaly viewing tool"""
+
+    # Form submits
+    calculate_submit = SubmitField()
+
+    # Generate dynamic lists of apertures to use in forms
+    aperture_list = []
+    for instrument in FULL_FRAME_APERTURES.keys():
+        for aperture in FULL_FRAME_APERTURES[instrument]:
+            item = [query_format(aperture), query_format(aperture)]
+            aperture_list.append(item)
+
+    miri_aperture_list = []
+    for aperture in FULL_FRAME_APERTURES['MIRI']:
+        miri_aperture_list.append([query_format(aperture), query_format(aperture)])
+
+    nirspec_aperture_list = []
+    for aperture in FULL_FRAME_APERTURES['NIRSPEC']:
+        nirspec_aperture_list.append([query_format(aperture), query_format(aperture)])
+    
+    nircam_aperture_list = []
+    for aperture in FULL_FRAME_APERTURES['NIRCAM']:
+        nircam_aperture_list.append([query_format(aperture), query_format(aperture)])
+
+    niriss_aperture_list = []
+    for aperture in FULL_FRAME_APERTURES['NIRISS']:
+        niriss_aperture_list.append([query_format(aperture), query_format(aperture)])
+
+    # Generate dynamic lists of filters to use in forms
+    filter_list = []
+    for instrument in FILTERS_PER_INSTRUMENT.keys():
+        # # if instrument in anomaly_query_config.INSTRUMENTS_CHOSEN:   # eg ['nirspec']: selects relevant filters, but not specific to chosen instruments
+        filters_per_inst = FILTERS_PER_INSTRUMENT[instrument]
+        for filter in filters_per_inst:
```

Comment:
It is really annoying, especially in our field of work, but `filter` is a python built-in function, and so we should avoid naming variables `filter`.  I would suggest to use `filt` instead (even though this violates the 'better to be explicit' rule -- as I said, really annoying 😅 )